sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs

```diff
@@ -46,6 +49,8 @@ public ServiceBusMessage(ReadOnlyMemory<byte> body)
         {
             AmqpMessageBody amqpBody = new AmqpMessageBody(new ReadOnlyMemory<byte>[] { body });
             AmqpMessage = new AmqpAnnotatedMessage(amqpBody);
+
+            this.body = new Lazy<BinaryData>(() => AmqpMessage.GetBody(), LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
Used the `LazyThreadSafetyMode.ExecutionAndPublication` to make sure when it throws the behavior is as before